src/sourceTracking.ts

```diff
@@ -106,6 +107,54 @@ export class SourceTracking extends AsyncCreatable {
     this.subscribeSDREvents = options.subscribeSDREvents ?? false;
   }
 
+  /**
+   * A static method to help delete custom labels from a file, or the entire file if there are no more labels
+   *
+   * @param filename - a path to a custom labels file
+   * @param customLabels - an array of SourceComponents representing the custom labels to delete
+   */
+  public static async deleteCustomLabels(filename: string, customLabels: SourceComponent[]): Promise<void> {
+    // for custom labels, we need to remove the individual label from the xml file
+    // so we'll parse the xml
+    const parser = new XMLParser({
+      ignoreDeclaration: false,
+      ignoreAttributes: false,
+      attributeNamePrefix: '@_',
+    });
+    const cls = parser.parse(fs.readFileSync(filename, 'utf8')) as {
```

Comment:
alternatively, there's `kit/ensureArray` so you can have fewer conditionals.   If you filtered the array and it ends up empty, you've got a check below for that to delete the file.